src/features/wisdom/wisdomOAuthProvider.ts

```diff
@@ -0,0 +1,458 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable tsdoc/syntax */
+import {
+  authentication,
+  AuthenticationProvider,
+  AuthenticationProviderAuthenticationSessionsChangeEvent,
+  AuthenticationSession,
+  Disposable,
+  env,
+  EventEmitter,
+  ExtensionContext,
+  ProgressLocation,
+  Uri,
+  window,
+  commands,
+} from "vscode";
+import { v4 as uuid } from "uuid";
+import { PromiseAdapter, promiseFromEvent } from "./utils";
+import axios from "axios";
+import { TreeDataProvider } from "./treeView";
+import {
+  generateCodeVerifier,
+  generateCodeChallengeFromVerifier,
+  UriEventHandler,
+  OAuthAccount,
+  calculateTokenExpiryTime,
+  WISDOM_AUTH_ID,
+  WISDOM_AUTH_NAME,
+  SESSIONS_SECRET_KEY,
+  ACCOUNT_SECRET_KEY,
+} from "./oAuthUtils";
+
+const CLIENT_ID = `XS8by8ctrzsk8GRJUyNLgT9GyuTNTs50MPexieew`; // cspell:disable-line
+
+const CODE_VERIFIER = generateCodeVerifier();
+const CODE_CHALLENGE = generateCodeChallengeFromVerifier(CODE_VERIFIER);
+const OAUTH_BASE_PATH = "http://127.0.0.1:8000";
```

Comment:
This should be picked from the base path and if it is empty/undefined we should throw and error.